File: Java/StudentGrade.java
import java.util.Arrays;
import java.util.Scanner;
public class StudentGrade{

	public static void main(String[]args){
		Scanner input = new Scanner(System.in);

		System.out.println("How many students do you have?");
		int studentSize = input.nextInt();

		System.out.println("How many subjects do they offer?");
		int subjectSize = input.nextInt();
		
		int [][] array = StudentGradeFunctions.saveSubjectAndStudentSize(subjectSize, studentSize);

		int score = 0;
		for(int counter = 0; counter < array.length; counter++){
			for(int index = 0; index < array[0].length; index++){
				System.out.printf("Entering score for student %d\n", counter+1);
				while(true){
					System.out.printf("Enter score for subject %d\n", index+1);
					score = input.nextInt();
					if(score>=0 && score<=100){
						array[counter][index]= score;
						break;
					}
				
			
			}
		System.out.println("\nSaving >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>");		
		System.out.println("Saved succesfully\n");	
		}
		System.out.println("=================================================================");
		

	}	

	int [] totalArray = StudentGradeFunctions.getTotalOfScore(array);
	System.out.print("STUDENT		"); 
	for(int index = 0; index < array[0].length; index++){
		System.out.printf("SUB%d	", index+1);
	}
	System.out.println("TOTAL	AVE	POS");
	System.out.println("=================================================================");

	for(int index = 0; index < array.length;index++){
	System.out.printf("Student%d", index+1);
		for(int count = 0; count < array[0].length; count++){
			System.out.printf("	%d",array[index][count]);
		}
	System.out.printf("	%d%n", totalArray[index]);
		
	}

	

	




















	}


}

File: oldSnacks/python/studentgrade.py


class_size = int(input("How many students do you have: "))
curriculum = int(input("How many subjects do you have: "))



"""if subject_size > 0 and student_size > 0:
		temp_subject_list = []
		for index in range(subject_size):
			temp_subject_list.append(0) 

		for index in range(student_size):
			list.append(temp_subject_list)

	list = [[0]*subject_size]* student_size"""

school_database = [[0 for subject in range(curriculum)] for student in range(class_size)]



for index in range(len(school_database)):
	for count in range(curriculum):
		while(True):
			print(f"Entering score for student {index+1}")
			print(f"Enter score for subject {count+1}")
			score = int(input(""))
			if score >=1 and score<=100:
				school_database[index][count] = score
				break
			else:
				print("\n<<SCORE MUST BE BETWEEN 1-100>>")
		
list_of_totals = []

for students in range(class_size):
	total = 0
	for subject in range(curriculum):
		total = total + school_database[students][subject]
	list_of_totals.append(total)
	


print("=========================================================================================")
print("STUDENT		", end="")
for index in range(curriculum):
	print(f"SUB{index+1}	", end="")
print("TOT	AVE	POS")

print("=========================================================================================")

for index in range(class_size):
	print(f"STUDENT{index+1}	", end="")
	for count in range(curriculum):
		print(f"{school_database[index][count]}	", end="")
	print(f"{list_of_totals[index]}	{list_of_totals[index]/curriculum}")

print("=========================================================================================")




